src/linear_algebra/finsupp.lean

```diff
@@ -141,247 +101,252 @@ begin
+theorem map_supported [inhabited α] (f : α → α') (s : set α) :
+  (supported β γ s).map (lmap_domain β γ f) = supported β γ (f '' s) :=
 begin
-  rcases exists_is_basis α β with ⟨s, hs⟩,
-  have : nonempty (fintype s),
-  { rwa [← cardinal.lt_omega_iff_fintype, hs.mk_eq_dim] },
-  cases this with hsf, letI := hsf,
-  calc cardinal.mk β = cardinal.mk (↥(lc.supported α s)) :
-    quotient.sound ⟨(module_equiv_lc hs).to_equiv⟩
-    ... = cardinal.mk (s →₀ α) :
-    begin
-      refine quotient.sound ⟨@linear_equiv.to_equiv α _ _ _ _ _ _ _ _⟩,
-      convert @lc.supported_equiv α β _ _ _ _ s,
-      { funext, exact subsingleton.elim _ _ },
-    end
-    ... = cardinal.mk (s → α) : quotient.sound ⟨finsupp.equiv_fun_on_fintype⟩
-    ... = _ : by rw [← hs.mk_eq_dim, cardinal.power_def]
+  refine le_antisymm (map_le_iff_le_comap.2 $
+    le_trans (supported_mono $ set.subset_preimage_image _ _)
+       (supported_comap_map _ _ _ _)) _,
+  intros l hl,
+  refine ⟨(lmap_domain β γ (function.inv_fun_on f s) : α' →₀ β →ₗ α →₀ β) l, λ x hx, _, _⟩,
+  { rcases finset.mem_image.1 (map_domain_support hx) with ⟨c, hc, rfl⟩,
+    exact function.inv_fun_on_mem (by simpa using hl hc) },
+  { rw [← linear_map.comp_apply, ← map_comp],
+    refine (map_domain_congr $ λ c hc, _).trans map_domain_id,
+    exact function.inv_fun_on_eq (by simpa using hl hc) }
 end
 
-lemma cardinal_lt_omega_of_dim_lt_omega
-  {α β : Type u} [discrete_field α] [add_comm_group β] [vector_space α β] [fintype α]
-  (h : dim α β < cardinal.omega) : cardinal.mk β < cardinal.omega :=
+theorem map_disjoint_ker (f : α → α') {s : set α}
+  (H : ∀ a b ∈ s, f a = f b → a = b) :
+  disjoint (supported β γ s) (lmap_domain β γ f).ker :=
 begin
-  rw [cardinal_mk_eq_cardinal_mk_field_pow_dim h],
-  exact cardinal.power_lt_omega (cardinal.lt_omega_iff_fintype.2 ⟨infer_instance⟩) h
+  rintro l ⟨h₁, h₂⟩,
+  rw [mem_coe, mem_ker, map_apply, map_domain] at h₂,
+  simp, ext x,
+  by_cases xs : x ∈ s,
+  { have : finsupp.sum l (λ a, finsupp.single (f a)) (f x) = 0, {rw h₂, refl},
+    rw [finsupp.sum_apply, finsupp.sum, finset.sum_eq_single x] at this,
+    { simpa [finsupp.single_apply] },
+    { intros y hy xy, simp [mt (H _ _ (h₁ hy) xs) xy] },
+    { simp {contextual := tt} } },
+  { by_contra h, exact xs (h₁ $ finsupp.mem_support_iff.2 h) }
 end
 
-end vector_space
+end lmap_domain
 
-namespace mv_polynomial
-universes u v
-variables {σ : Type u} {α : Type v} [decidable_eq σ]
+section total
+variables (α) {α' : Type*} (β) {β' : Type*} (γ)
+          [decidable_eq α'] [decidable_eq β'] [add_comm_group β'] [module γ β']
+          (v : α → β) {v' : α' → β'}
 
-instance [discrete_field α] : vector_space α (mv_polynomial σ α) :=
-finsupp.vector_space
+/-- Interprets (l : α →₀ γ) as linear combination of the elements in the family (v : α → β) and
+    evaluates this linear combination. -/
+protected def total : α →₀ γ →ₗ β := finsupp.lsum (λ i, linear_map.id.smul_right (v i))
```

Comment:
alpha and beta can be implicit here

Reply:
In the expression `finsupp.total α β γ v l`, it is very convenient to have the coersion of the linear map `finsupp.total α β γ v` into a function. I don't understand why, but if I make alpha and beta implicit, the coersion does no longer work. Any ideas?